<?xml version="1.0" ?>
<robot name="Assembly" xmlns:xacro="http://www.ros.org/wiki/xacro">

  <xacro:property name="file_dir" value="file:///home/zaid-khan/urdf"/>

  <material name="silver">
    <color rgba="0.3 0.3 0.3 1.000"/>
  </material>

  <material name="black">
    <color rgba="0.1 0.1 0.1 1.000"/>
  </material>

  <material name="grey">
    <color rgba="0.01 0.01 0.01 1.000"/>
  </material>

  <link name="base_link">
    <inertial>
      <origin xyz="0.00013823687262010676 -1.287275748997914e-14 0.030999685894238568" rpy="0 0 0"/>
      <mass value="3.85463310643452"/>
      <inertia ixx="0.006744" iyy="0.006727" izz="0.007308" ixy="0.0" iyz="-0.0" ixz="1e-06"/>
    </inertial>
    <visual>
      <origin xyz="0 0 0" rpy="0 0 0"/>
      <geometry>
        <mesh filename="${file_dir}/meshes/base_link.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="silver"/>
    </visual>
    <collision>
      <origin xyz="0 0 0" rpy="0 0 0"/>
      <geometry>
        <mesh filename="${file_dir}/meshes/base_link.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>

  <link name="linkone_1">
    <inertial>
      <origin xyz="0.0486628984025515 2.0300039683640878e-15 0.06383563728609049" rpy="0 0 0"/>
      <mass value="2.363220430083907"/>
      <inertia ixx="0.006563" iyy="0.007612" izz="0.001802" ixy="-0.0" iyz="-0.0" ixz="-0.001539"/>
    </inertial>
    <visual>
      <origin xyz="-0.0 0.0 -0.09" rpy="0 0 0"/>
      <geometry>
        <mesh filename="${file_dir}/meshes/linkone_1.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="silver"/>
    </visual>
    <collision>
      <origin xyz="-0.0 0.0 -0.09" rpy="0 0 0"/>
      <geometry>
        <mesh filename="${file_dir}/meshes/linkone_1.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>

  <link name="linktwo_1">
    <inertial>
      <origin xyz="-0.037500000000000144 -0.044246151650264166 0.043397671320847375" rpy="0 0 0"/>
      <mass value="6.122340532641328"/>
      <inertia ixx="0.016208" iyy="0.009234" izz="0.013673" ixy="0.0" iyz="0.005008" ixz="0.0"/>
    </inertial>
    <visual>
      <origin xyz="-0.0475 0.0 -0.24" rpy="0 0 0"/>
      <geometry>
        <mesh filename="${file_dir}/meshes/linktwo_1.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="silver"/>
    </visual>
    <collision>
      <origin xyz="-0.0475 0.0 -0.24" rpy="0 0 0"/>
      <geometry>
        <mesh filename="${file_dir}/meshes/linktwo_1.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>

  <link name="linkthree_1">
    <inertial>
      <origin xyz="-0.022500000000000166 -0.053128104103918355 -0.050135367274481035" rpy="0 0 0"/>
      <mass value="3.5193373539764488"/>
      <inertia ixx="0.012798" iyy="0.010792" izz="0.00319" ixy="-0.0" iyz="-0.002538" ixz="0.0"/>
    </inertial>
    <visual>
      <origin xyz="-0.033 0.129904 -0.315" rpy="0 0 0"/>
      <geometry>
        <mesh filename="${file_dir}/meshes/linkthree_1.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="silver"/>
    </visual>
    <collision>
      <origin xyz="-0.033 0.129904 -0.315" rpy="0 0 0"/>
      <geometry>
        <mesh filename="${file_dir}/meshes/linkthree_1.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>

  <link name="end_effector_1">
    <inertial>
      <origin xyz="-0.010000000000000179 1.8935329859637307e-07 -0.038706393001661066" rpy="0 0 0"/>
      <mass value="0.71040762907869"/>
      <inertia ixx="0.000248" iyy="0.001263" izz="0.001142" ixy="0.0" iyz="-0.0" ixz="0.0"/>
    </inertial>
    <visual>
      <origin xyz="-0.0205 0.197404 -0.1575" rpy="0 0 0"/>
      <geometry>
        <mesh filename="${file_dir}/meshes/end_effector_1.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="grey"/>
    </visual>
    <collision>
      <origin xyz="-0.0205 0.197404 -0.1575" rpy="0 0 0"/>
      <geometry>
        <mesh filename="${file_dir}/meshes/end_effector_1.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>

  <link name="gripper_right_1">
    <inertial>
      <origin xyz="-0.006927083333333459 1.8965585155128473e-07 -0.03044270848234537" rpy="0 0 0"/>
      <mass value="0.2512"/>
      <inertia ixx="7.9e-05" iyy="8.7e-05" izz="2.5e-05" ixy="0.0" iyz="0.0" ixz="-4e-06"/>
    </inertial>
    <visual>
      <origin xyz="-0.0905 0.197404 -0.1075" rpy="0 0 0"/>
      <geometry>
        <mesh filename="${file_dir}/meshes/gripper_right_1.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="black"/>
    </visual>
    <collision>
      <origin xyz="-0.0905 0.197404 -0.1075" rpy="0 0 0"/>
      <geometry>
        <mesh filename="${file_dir}/meshes/gripper_right_1.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>

  <link name="gripper_left_1">
    <inertial>
      <origin xyz="0.006927083333333153 1.8965585160679588e-07 -0.030442708482345426" rpy="0 0 0"/>
      <mass value="0.2512"/>
      <inertia ixx="7.9e-05" iyy="8.7e-05" izz="2.5e-05" ixy="0.0" iyz="-0.0" ixz="4e-06"/>
    </inertial>
    <visual>
      <origin xyz="0.0695 0.197404 -0.1075" rpy="0 0 0"/>
      <geometry>
        <mesh filename="${file_dir}/meshes/gripper_left_1.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="black"/>
    </visual>
    <collision>
      <origin xyz="0.0695 0.197404 -0.1075" rpy="0 0 0"/>
      <geometry>
        <mesh filename="${file_dir}/meshes/gripper_left_1.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>

  <joint name="Revolute_1" type="continuous">
    <origin xyz="0.0 0.0 0.09" rpy="0 0 0"/>
    <parent link="base_link"/>
    <child link="linkone_1"/>
    <axis xyz="0.0 0.0 -1.0"/>
  </joint>

  <joint name="Revolute_2" type="revolute">
    <origin xyz="0.0475 0.0 0.15" rpy="0 0 0"/>
    <parent link="linkone_1"/>
    <child link="linktwo_1"/>
    <axis xyz="-1.0 0.0 0.0"/>
    <limit upper="1.07" lower="-0.36" effort="100" velocity="100"/>
  </joint>

  <joint name="Revolute_3" type="revolute">
    <origin xyz="-0.0145 -0.129904 0.075" rpy="0 0 0"/>
    <parent link="linktwo_1"/>
    <child link="linkthree_1"/>
    <axis xyz="-1.0 0.0 0.0"/>
    <limit upper="1.57" lower="-0.1" effort="100" velocity="100"/>
  </joint>

  <joint name="Revolute_4" type="revolute">
    <origin xyz="-0.0125 -0.0675 -0.1575" rpy="0 0 0"/>
    <parent link="linkthree_1"/>
    <child link="end_effector_1"/>
    <axis xyz="-1.0 0.0 0.0"/>
    <limit upper="1.57" lower="-0.55" effort="100" velocity="100"/>
  </joint>

  <joint name="Slider_1" type="prismatic">
    <origin xyz="0.07 0.0 -0.05" rpy="0 0 0"/>
    <parent link="end_effector_1"/>
    <child link="gripper_right_1"/>
    <axis xyz="1.0 -0.0 -0.0"/>
    <limit upper="-0.01" lower="-0.06" effort="100" velocity="100"/>
  </joint>

  <joint name="Slider_2" type="prismatic">
    <origin xyz="-0.09 0.0 -0.05" rpy="0 0 0"/>
    <parent link="end_effector_1"/>
    <child link="gripper_left_1"/>
    <axis xyz="-1.0 0.0 0.0"/>
    <limit upper="-0.01" lower="-0.06" effort="100" velocity="100"/>
  </joint>

</robot>
